File: sudoq-app/sudoqapp/src/main/java/de/sudoq/controller/menus/SudokuTypeOrder.java
package de.sudoq.controller.menus;

import java.util.Comparator;

import de.sudoq.model.sudoku.sudokuTypes.SudokuTypes;

public class SudokuTypeOrder {

    public static int getKey(SudokuTypes s){
        return ordinals[s.ordinal()];
    }

    private static int[] ordinals;
    static{

        SudokuTypes[] order = {SudokuTypes.standard9x9,
                               SudokuTypes.standard4x4,
                               SudokuTypes.standard6x6,
                               SudokuTypes.standard16x16,
                               SudokuTypes.samurai,
                               SudokuTypes.Xsudoku,
                               SudokuTypes.HyperSudoku,
                               SudokuTypes.squigglya,
                               SudokuTypes.squigglyb,
                               SudokuTypes.stairstep};

        ordinals = new int[SudokuTypes.values().length];

        for (int i = 0; i < order.length; i++) {
            ordinals[order[i].ordinal()] = i;
        }

    }

}

File: sudoq-app/sudoqapp/src/main/kotlin/de/sudoq/controller/menus/SudokuTypeOrder.kt
package de.sudoq.controller.menus

import de.sudoq.model.sudoku.sudokuTypes.SudokuTypes

object SudokuTypeOrder {

    //for a sudoku type, get its position in the drop down menu
    fun getKey(s: SudokuTypes): Int {
        return ordinals[s.ordinal]
    }

    //type enum index -> position in ui
    private var ordinals: IntArray

    init {
        // position in ui -> enum
        val order = arrayOf(
            SudokuTypes.standard9x9,
            SudokuTypes.standard4x4,
            SudokuTypes.standard6x6,
            SudokuTypes.standard16x16,
            SudokuTypes.samurai,
            SudokuTypes.Xsudoku,
            SudokuTypes.HyperSudoku,
            SudokuTypes.squigglya,
            SudokuTypes.squigglyb,
            SudokuTypes.stairstep
        )
        //enum id -> ui pos
        ordinals = IntArray(SudokuTypes.entries.size)
        for ((uiPos, enum) in order.withIndex()) {
            ordinals[enum.ordinal] = uiPos
        }
    }
}